src/Components/Authentication/ForgotPasswordPage.jsx

```diff
@@ -0,0 +1,265 @@
+import { useReducer, useState } from "react";
+import Loader from "react-loader-spinner";
+import { Link } from "react-router-dom";
+
+import { InputPasswordField } from "./InputPasswordField";
+import { useAuthentication } from "../../context";
+
+export const ForgotPasswordPage = () => {
+  const [error, setError] = useState("");
+  const [isLoading, setLoading] = useState(false);
+  const [showSuccess, setSuccess] = useState(false);
+  const { updateUserDetails } = useAuthentication();
+
+  const initialState = {
+    username: "",
+    password: "",
+    confirmPassword: ""
+  };
+
+  const formReducer = (state, { type, payload }) => {
+    switch (type) {
+      case "SET_USERNAME":
+        return { ...state, username: payload };
+
+      case "SET_PASSWORD":
+        return { ...state, password: payload };
+
+      case "SET_RE_PASSWORD":
+        return { ...state, confirmPassword: payload };
+
+      default:
+        return state;
+    }
+  };
+
+  const [formState, formDispatch] = useReducer(formReducer, initialState);
+
+  const errorsInitialState = {
+    usernameError: "",
+    passwordError: "",
+    confirmPasswordError: ""
+  };
+
+  const errorReducer = (state, { type, payload }) => {
+    switch (type) {
+      case "SET_USERNAME_ERROR":
+        return { ...state, usernameError: payload };
+
+      case "SET_PASSWORD_ERROR":
+        return { ...state, passwordError: payload };
+
+      case "SET_RE_PASSWORD_ERROR":
+        return { ...state, confirmPasswordError: payload };
+
+      case "CLEAR_ERRORS_ERROR":
+        return { errorsInitialState };
+
+      default:
+        return state;
+    }
+  };
+
+  const [fieldErrors, errorsDispatch] = useReducer(
+    errorReducer,
+    errorsInitialState
+  );
+
+  const onFocusClearError = (type) => {
+    errorsDispatch({ type, payload: "" });
+  };
+
+  const checkFormValidity = () => {
+    let errorFlag = true;
+
+    if (formState.username === "" || !/^.+@.+\.com$/.test(formState.username)) {
+      errorsDispatch({

```

Comment:
seems like a email check, but then why variable name is username ? Can you please rename it to email for proper understanding!

Reply:
Done